modules/module-mysql/test/src/MysqlTypeMappings.test.ts

```diff
@@ -0,0 +1,302 @@
+    multipolygon_col MULTIPOLYGON,
+    geometrycollection_col GEOMETRYCOLLECTION
+    )`);
+
+    connection.release();
+  }
+
+  test('Number types mappings', async () => {
+    await setupTable();
+    await connectionManager.query(`
+INSERT INTO test_data (
+    tinyint_col,
+    smallint_col,
+    mediumint_col,
+    int_col,
+    integer_col,
+    bigint_col,
+ -- float_col,
+    double_col,
+    decimal_col,
+    numeric_col,
+    bit_col,
+    boolean_col
+    -- serial_col is auto-incremented and can be left out
+) VALUES (
+    127,                 -- TINYINT maximum value
+    32767,               -- SMALLINT maximum value
+    8388607,             -- MEDIUMINT maximum value
+    2147483647,          -- INT maximum value
+    2147483647,          -- INTEGER maximum value
+    9223372036854775807, -- BIGINT maximum value
+    -- 3.14,             -- FLOAT example
+    3.1415926535,        -- DOUBLE example
+    12345.67,            -- DECIMAL(10,2) example
+    12345.67,            -- NUMERIC(10,2) example
+    b'10101010',         -- BIT(8) example in binary notation
+    TRUE                 -- BOOLEAN value (alias for TINYINT(1))
+    -- serial_col is auto-incremented
+)`);
+
+    const databaseRows = await getDatabaseRows(connectionManager, 'test_data');
+    const replicatedRows = await getReplicatedRows();
+
+    const expectedResult = {
+      tinyint_col: 127,
+      smallint_col: 32767,
+      mediumint_col: 8388607,
+      int_col: 2147483647,
+      integer_col: 2147483647,
+      bigint_col: 9223372036854775807n,
+      //float_col: 3.14,
+      double_col: 3.1415926535,
+      decimal_col: 12345.67,
+      numeric_col: 12345.67,
+      bit_col: new Uint8Array([0b10101010]).valueOf(),
+      boolean_col: 1,
+      serial_col: 1
+    };
+    expect(databaseRows[0]).toMatchObject(expectedResult);
+    expect(replicatedRows[0]).toMatchObject(expectedResult);
+  });
+
+  test('Character types mappings', async () => {
+    await setupTable();
+    await connectionManager.query(`
+INSERT INTO test_data (
+    char_col,
+    varchar_col,
+    binary_col,
+    varbinary_col,
+    tinyblob_col,
+    blob_col,
+    mediumblob_col,
+    longblob_col,
+    tinytext_col,
+    text_col,
+    mediumtext_col,
+    longtext_col,
+    enum_col,
+    set_col
+) VALUES (
+    'CharData',               -- CHAR(10) with padding spaces
+    'Variable character data',-- VARCHAR(255)
+    'ShortBin',        -- BINARY(16)
+    'VariableBinaryData',     -- VARBINARY(256)
+    'TinyBlobData',           -- TINYBLOB
+    'BlobData',               -- BLOB
+    'MediumBlobData',         -- MEDIUMBLOB
+    'LongBlobData',           -- LONGBLOB
+    'TinyTextData',           -- TINYTEXT
+    'TextData',               -- TEXT
+    'MediumTextData',         -- MEDIUMTEXT
+    'LongTextData',           -- LONGTEXT
+    'value1',                 -- ENUM('value1', 'value2', 'value3')
+    'value1,value3'           -- SET('value1', 'value2', 'value3')
```

Comment:
Should we perhaps represent this as a JSON array, i.e. `["value1","value3"]`? Not 100% convinced that's the best approach, just wondering if that would be easier to work with. For example, our `IN` operator works with JSON arrays automatically.

Reply:
Oh yeah I think it would be. Let me take look what that mapping entails.